CONTRIBUTING.md

```diff
@@ -0,0 +1,137 @@
+a _drop-in replacement_ for `flag` (and then some of course, but at least that).
+
+### Versioning
+
+This package abides by `semver`.
+
+Hence:
+* `v1.0.6` : patch
+* `v1.1.0` : new feature
+
+As stated above, it is unlikely we ever issue a `v2.0.0` (breaking change).
+
+## Pull requests
+
+All contributions are much appreciated and welcome. There is no small contributions: additional documentation, examples, 
+tests, etc are all valuable additions, not just new features.
+
+### Linting
+
+This repository uses `github.com/golangci/golangci-lint/cmd/golangci-lint` as meta-linter, enforced by CI.
+The set of rules may evolve over time. `revive` and `govet` are must-have.
+
+> TODO: agree on the initial .golangci.yml setup, with a no-nonsense approach to linting. Code quality without nitpicking... :)
+> As a started, here is a sample configuration that I use on a daily basis for personal projects (no strong agenda on this):
+> [golangci-lint.yml](https://github.com/fredbi/gflag/blob/master/.golangci.yml).
+
+### Testing
+
+Pull requests should include significant unit tests.
+
+This repository uses `github.com/stretchr/testify` framework for unit tests.
+
+> TODO: to be discussed, what we mean by "significant", e.g. >70% test coverage or so.
+
+### Documentation
+
+Keeping a compelling `godoc` documentation is part of the "drop-in replacement" promise.
+
+Any newly exposed API should come with a clear, readable description of what it does.
+
+Testable example are alway welcome (although not required).
+
+Conversely, internals only need comments if a particular caveat or expectation requires wording besides just code.
+
+### Dependencies
+
+So far, `pflag` has managed to build without _any_ external dependency.
+
+We may indulge into _a few_ dependencies, especially test dependencies, or things like `golang.org/x/...`.
+
+We should not need much more than that, unless we are talking about extensions (e.g. new flag types). See below.
+
+### Commit etiquette
+
+Please keep your commit title and description concise and to the point.
+
+Feel free to `git rebase -i`, `git commit --amend` etc, so this repository keeps a nice and readable git history,
+without merge commits or fixup rescindments.
+
+### Contributor Licence Agreement
+
+Contributions are subject to a CLA DCO being signed off.
+
+> TODO: is this something that is still needed? Anyhow, check if the CLA terms are up to date.
+> The Linux Foundation has introduced [EasyCLA](https://docs.linuxfoundation.org/lfx/easycla).
+> I believe that `cla.assistant.io` is a bit old.
+
+### Signed commits
+
+Commits shall be [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).
+
+> TODO: this requirement is debatable.
+> On the one hand, it is certainly something many junior developers are perplex about.
+> On the other hand, supporting so many major pieces of infrastructure is a big responsibility.
```

Comment:
I will very often make a PR via the Github UI - for doc changes or minor cosmetic issues. I think its possible to sign in the UI, but I very frequently forget to, and then its a pain to come back and fix it locally (sometimes to the point of me just closing the PR and saying someone else can re-contribute it).

Reply:
@josegonzalez  this is simple sign-off (git commit -s, with email) not GPG signature.